### DISCUSSION TITLE
thread/condition

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#294] [Announcements]
# thread/condition

Condition是Java并发工具包java.util.concurrent.locks中的一个接口，它提供了一种更灵活的线程同步机制。与Object类中的wait(), notify(), 和 notifyAll()方法相比，Condition为多线程间的协调和通信提供了更强大的功能。

https://javabetter.cn/thread/condition.html

**Comment by Ricehahaa** [comment]
signal/signalAll 实现原理那里写错了吧
”1.将头节点的状态更改为 CONDITION；“ ，不应该是”1.将头节点的状态更改为0；“吗

**Reply by Liucc-123** [reply]
是的，应该是写错了

**Comment by LFAW233** [comment]
ConditionObject怎么是AQS的子类啊，明明是内部类啊

**Reply by LFAW233** [reply]
但实话实说这章的讲解比前几章好多了，整个逻辑清清楚楚

**Reply by Liucc-123** [reply]
是的  同感！这个章节的绘图把流程解释的就很直观清晰！
